packages/arb-token-bridge-ui/src/hooks/useBalances.ts

```diff
@@ -0,0 +1,57 @@
+import { Address, useAccount } from 'wagmi'
+
+import { useBalance } from './useBalance'
+import { useNetworks } from './useNetworks'
+import { useNetworksRelationship } from './useNetworksRelationship'
+
+import { useDestinationAddressStore } from '../components/TransferPanel/AdvancedSettings'
+
+export function useBalances({
+  parentWalletAddress,
+  childWalletAddress
+}: {
+  parentWalletAddress?: Address
+  childWalletAddress?: Address
+} = {}) {
+  const [networks] = useNetworks()
+  const { childChain, parentChain, isDepositMode } =
+    useNetworksRelationship(networks)
+  const { address: walletAddress } = useAccount()
+  const { destinationAddress } = useDestinationAddressStore()
+  const destinationAddressOrWalletAddress = destinationAddress || walletAddress
+
+  const _parentWalletAddress =
+    parentWalletAddress ??
+    (isDepositMode ? walletAddress : destinationAddressOrWalletAddress)
+
+  const _childWalletAddress =
+    childWalletAddress ??
+    (isDepositMode ? destinationAddressOrWalletAddress : walletAddress)
+
+  const {
+    eth: [ethParentBalance, updateEthParentBalance],
+    erc20: [erc20ParentBalances, updateErc20ParentBalances]
+  } = useBalance({
+    chainId: parentChain.id,
+    walletAddress: _parentWalletAddress
+  })
+
+  const {
+    eth: [ethChildBalance, updateEthChildBalance],
+    erc20: [erc20ChildBalances, updateErc20ChildBalances]
+  } = useBalance({
+    chainId: childChain.id,
+    walletAddress: _childWalletAddress
+  })
+
+  return {
+    ethParentBalance,
+    updateEthParentBalance,
+    erc20ParentBalances,
+    updateErc20ParentBalances,
+    ethChildBalance,
+    updateEthChildBalance,
+    erc20ChildBalances,
+    updateErc20ChildBalances
+  }
```

Comment:
It's not an issue if it's used as `const { x, y } = useBalances()` but if they used the entire object, it should be memoed